Tests/SonarQube.TeamBuild.PreProcessor.Tests/Infrastructure/MockSonarQubeServer.cs

```diff
@@ -40,21 +40,12 @@ public void AssertMethodCalled(string methodName, int callCount)
 
         IEnumerable<string> ISonarQubeServer.GetActiveRuleKeys(string qualityProfile, string language, string repository)
         {
-            if (string.IsNullOrWhiteSpace(qualityProfile))
-            {
-                throw new ArgumentNullException("qualityProfile");
-            }
-            if (string.IsNullOrWhiteSpace(language))
-            {
-                throw new ArgumentNullException("language");
-            }
-            if (string.IsNullOrWhiteSpace(repository))
-            {
-                throw new ArgumentNullException("repository");
-            }
-
             this.LogMethodCalled();
 
+            Assert.IsFalse(string.IsNullOrEmpty(qualityProfile), "Quality profile is required");
+            Assert.IsFalse(string.IsNullOrEmpty(language), "Language is required");
+            Assert.IsFalse(string.IsNullOrEmpty(repository), "Repository is required");
+
```

Comment:
Minor standarisation: all of the test methods now log that the call was made then assert the values are not null. Previously some methods threw exceptions and some asserted.